### PR TITLE
Added User model and authentication routes

### LINKED ISSUE #21: User profile -Backend
- [x] backend / API support to get user info

### PR DESCRIPTION
**Pull request by shivanshs9**
Fixes #21 

- [x] Backend routes for authentication and user model
- [ ] Frontend page for User profile & Authentication (according to Figma)

### CODE REVIEW

**Comment by shivanshs9** [comment]
It's a draft PR till I actually test the auth functionality... However, the website doesn't work as mentioned in the slack group.

File: backend/models/location.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const { Schema, model } = require('mongoose')
+
+// User Schema
+const LocationSchema = new Schema({
```
Comment (robinv85) [inline backend/models/location.js:4]: we could store `{ latitude, longitude }` from the html5 geo api, on the user
and when we retrieved country, city, zip also store that

File: backend/models/user.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const { Schema, model } = require('mongoose')
+
+// User Schema
+const UserSchema = new Schema({
+  firstName: {
+    type: String,
+    required: true,
```
[the remark this answers is not in the document]
Comment (shivanshs9) [reply backend/models/user.js:7]: Auth0 provides name info, that's why it's required.

File: backend/models/user.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    required: false,
+  },
+  dateJoined: {
+    type: Date,
+    default: Date.now,
+  },
+  location: {
+    type: Schema.Types.ObjectId,
+    ref: 'Location'
+  }
```
[the remark this answers is not in the document]
Comment (shivanshs9) [reply backend/models/user.js:29]: There's also a way to register without answering, right? So it has to be optional I guess

[the remark this answers is not in the document]
Comment (robinv85) [reply backend/models/user.js:29]: yep true it's best to keep the onboarding flow as easy as possible with the least required info

**Pull request closed without merging**